ocaml/libs/clock/date.ml

```diff
@@ -60,30 +56,39 @@ let best_effort_iso8601_to_rfc3339 x =
   in
   match tz with
   | None | Some "" ->
-      (* the caller didn't specify a tz. we must try to add one so that ptime can at least attempt to parse *)
-      (Printf.sprintf "%sZ" x, Empty)
-  | Some tz ->
-      (x, TZ tz)
+      (* the caller didn't specify a tz, assume Universal Local Time *)
+      Printf.sprintf "%sZ" x
+  | Some _ ->
+      x
 
 let of_iso8601 x =
-  let rfc3339, print_timezone = best_effort_iso8601_to_rfc3339 x in
+  let rfc3339 = best_effort_iso8601_to_rfc3339 x in
   match Ptime.of_rfc3339 rfc3339 |> Ptime.rfc3339_error_to_msg with
   | Error _ ->
       invalid_arg (Printf.sprintf "%s: %s" __FUNCTION__ x)
-  | Ok (t, tz, _) -> (
-    match tz with
-    | None | Some 0 ->
-        Ptime.to_date_time t |> of_dt print_timezone
-    | Some _ ->
-        invalid_arg (Printf.sprintf "%s: %s" __FUNCTION__ x)
-  )
-
-let to_rfc3339 ((y, mon, d), ((h, min, s), _), print_type) =
-  match print_type with
-  | TZ tz ->
-      Printf.sprintf "%04i%02i%02iT%02i:%02i:%02i%s" y mon d h min s tz
-  | Empty ->
-      Printf.sprintf "%04i%02i%02iT%02i:%02i:%02i" y mon d h min s
+  | Ok (t, tz, _) ->
+      {t; tz}
+
+let print_tz tz_s =
+  match tz_s with
+  | None ->
+      ""
+  | Some 0 ->
+      "Z"
+  | Some tz ->
+      let tz_sign = if tz < 0 then '-' else '+' in
+      let all_tz_minutes = tz / 60 in
```

Comment:
`abs` could be applied here once?

Reply:
I believe so, I've a test with negative timezone.